lib/galaxy/tools/deps/__init__.py

```diff
@@ -192,13 +192,17 @@ def dependency_shell_commands( self, requirements, **kwds ):
         """
         Runs a set of requirements through the dependency resolvers and returns
         a list of commands required to activate the dependencies. If dependencies
-        are cacheable and the cache exists, will generate commands to activate
-        cached environments.
+        are cacheable and the cache does not exist, will try to create it.
+        If cached environment exists or is successfully created, will generate
+        commands to activate it.
         """
         resolved_dependencies = self.requirements_to_dependencies(requirements, **kwds)
         cacheable_dependencies = [dep for dep in resolved_dependencies.values() if dep.cacheable]
         hashed_dependencies_dir = self.get_hashed_dependencies_path(cacheable_dependencies)
-        if os.path.exists(hashed_dependencies_dir):
+        if not os.path.exists(hashed_dependencies_dir):
```

Comment:
Could you add a check for `conda_auto_install` here? I am very much afraid of triggering this at random (whenever these dependencies are being resolved for the first time...) timepoints on production instances.